screens/Menu/NearbyInterest.js

```diff
@@ -19,14 +20,49 @@ import PropTypes from "prop-types";
  * @param  {} navigation props.navigation is the name of the object from Navigator library
  */
 function NearbyInterest(props) {
+   
+    const [fromScreen, setFromScreen] = React.useState();
+    /**
+     * The asyncstorage getter that will let us grab the value coming from the bottomMenu component
+     * @param  {} =>{letname=awaitAsyncStorage.getItem("sideMenu)"
+     * @param  {} ;setFromScreen(name)
+     */
+    const getFromScreen = async () => {
+        let name = await AsyncStorage.getItem("sideMenu");
+        setFromScreen(name);
+    };
+
+    useEffect(() => {
```

Comment:
Can we avoid useEffects & AsyncStorage and use props.navigation here?

Reply:
The original plan was to absolutely not use the AsyncStorage and the useEffects, but my hand was force because with props only, without using useEffect, there was no way to re-rendered the screen.

@AlainJobs  and I had a hour long discussion over it, the complexity comes from the fact that we use the navigation between two screens that are not in the same stack, meaning that the states of the components are never re-rendered, and the states of the hooks are never refreshed. (You would come back to NearbyInterest in the state of the hook would be the same as the previous time you had access the screen)

It would of been possible to re-render the screens if we could of use the goBack() method to pop out the screen from the stacks, but the goBack() does not work in this situation because screens **can only go back** to screens that are within the same stack, in this they are not.

Thus, since we could not use props, the final alternative was to use useEffecs & AsyncStorage